src/dataset.jl

```diff
@@ -172,13 +172,114 @@ end
     write(dataset::AbstractDataset, filename::AbstractString; kwargs...)
 
 Writes the dataset to the designated filename.
+
+### Parameters
+* `dataset`: The dataset to write
+* `filename`: The filename, UTF-8 encoded.
+
+### Keyword Arguments
+* `driver`: The driver to use, you have to manually select the right driver via `getdriver(drivername)` matching the file extension you wish. 
+Otherwise the driver of the source dataset will be used.
+* `options` (Vector{String}): A vector of strings containing KEY=VALUE pairs for driver-specific creation options.
+* `layer_options` (Vector{String}): Driver specific options for layer creation when writing vector datasets.
+* `chunksize` (Integer): Number of features to write in one database transaction. Neglected when `use_gdal_copy` is true.
+* `use_gdal_copy` (Bool): Set this to `true` if you encounter errors when writing. Set to `false` for higher speed.
+* `strict` (Bool): Set this to `true` if the written dataset should be a 1:1 copy of the source data, default is `false`,
+which allows the driver to adapt if necessary.
+    
+### Returns
+`nothing`
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
-    destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    if nraster(dataset) > 0 && nlayer(dataset) > 0
+        error("Writing datasets with raster and vector data is not supported.")
```

Comment:
If we hit this case, can we print out the driver etc and encourage them to file an issue in this package with the printed information?

Reply:
Yup, I'll change the info.